content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md

```diff
@@ -0,0 +1,82 @@
+# The time is generally irrelvant in how Valkey published, so '01:01:01' is a good placeholder
+date= 2024-07-08 01:01:01
+# 'description' is what is shown as a snippet/summary in various contexts.
+# You can make this the first few lines of the post or (better) a hook for readers.
+# Aim for 2 short sentences.
+description= "Better efficiency reduces cost, improves latency and makes our environment greener"
+# 'authors' are the folks who wrote or contributed to the post.
+# Each author corresponds to a biography file (more info later in this document)
+authors= [ "dantouitou", "uriyagelnik"]
++++
+Valkey, the popular open-source key-value store that succeeded OSS Redis, captures the imagination of developers around the world.
+From simple in-memory caching implementations to complex job queues, real-time collaboration, and leaderboard applications, we at AWS are continually amazed by how innovatively users employ Valkey.
+Clearly, this is just the tip of the iceberg.
+As more use cases and industries seek to benefit from the speed, low latency, and cost reduction advantages of in-memory processing as introduced by Valkey, we are fully committed to our vision of delivering exceptional performance.
+
+### Our Commitment to performance and efficiency
+At AWS, we have hundreds of thousands of customers using Amazon ElastiCache and Amazon MemoryDB.
+Feedback we continuously hear from customers is that they need better absolute performance and squeeze more performance from their clusters.
+
+For absolute performance per node, customers sometimes have hot keys or other limitation that prevents them from scaling out.
+They also need better latencies at high percentiles.
+For efficiency, we see customers trying to optimize their costs and get more out of their boxes.
+
+Besides performance, we also see efficiency as a way to improve resilience.
+For example, with fork-based snapshotting and full-sync replication, the longer it takes, the more memory is consumed by copied-on-write pages and the replica output buffer, often resulting in high swapping and erratic behavior.
+By reducing execution time, one can achieve a more predictable memory usage and greater stability.
+
+### Major Upgrade to Valkey Performance
+
+We are excited to contribute  our major performance improvements and expertise to Valkey.
+Starting with version 8, Valkey users will benefit from a significant increase in performance, thanks to a new multi-threading implementation that can boost throughput and latency on a wide range of hardware types.
+
+![I/O threads high level design](/assets/media/pictures/performance_comparison.png)
+
+The data demonstrates a substantial performance improvement with the new I/O threads approach.
+Throughput increased by approximately 230%, rising from 360K to 1.19M requests per second.
+Latency metrics improved across all percentiles, with average latency decreasing by 69.8% from 1.792 ms to 0.542 ms. 
+### Benefits of High Capacity Shards
```

Comment:
I think this paragraph is good, but just in the wrong place. I think we should move it up above `### Major Upgrade to Valkey Performance`.